src/program.ts

```diff
@@ -0,0 +1,286 @@
+   */
+  constructor(
+    public connection: Connection,
+    public payerAccount: Keypair,
+    public programAccount: Keypair
+  ) {
+    this.logs = new LogsParser(this.programAccount.publicKey, this.connection);
+  }
+
+  /**
+   * Create a storage account
+   *
+   * @param space
+   * @returns
+   */
+  public async createStorageAccount(space: number): Promise<Keypair> {
+    const lamports = await this.connection.getMinimumBalanceForRentExemption(
+      space
+    );
+
+    let account = Keypair.generate();
+
+    const transaction = new Transaction().add(
+      SystemProgram.createAccount({
+        fromPubkey: this.payerAccount.publicKey,
+        newAccountPubkey: account.publicKey,
+        lamports,
+        space,
+        programId: this.programAccount.publicKey,
+      })
+    );
+
+    await sendAndConfirmTransaction(
+      this.connection,
+      transaction,
+      [this.payerAccount, account],
+      {
+        skipPreflight: false,
+        commitment: 'confirmed',
+        preflightCommitment: undefined,
+      }
+    );
+
+    return account;
+  }
+
+  /**
+   * Invokes the given callback on every program log.
+   *
+   * @param callback  The function to invoke whenever a `Program log:` is parsed in the logs.
+   */
+  public addLogListener(callback: LogCallback): number {
+    return this.logs.addLogListener(callback);
+  }
+
+  /**
+   * Unsubscribes from the given log listener id.
+   *
+   * @param listenerId The log listener id
+   */
+  public async removeLogListener(listenerId: number): Promise<void> {
+    return await this.logs.removeLogListener(listenerId);
+  }
+
+  /**
+   * Invokes the given callback every time the given event is emitted.
+   *
+   * @param abi       ABI interface
+   * @param callback  The function to invoke whenever the event is emitted from
+   *                  program logs.
+   */
+  public addEventListener(
+    abi: ethers.utils.Interface,
+    callback: EventCallback
+  ): number {
+    return this.logs.addEventListener(abi, callback);
+  }
+
+  /**
+   * Unsubscribes from the given eventName.
+   *
+   * @param listenerId The log listener id
+   */
+  public async removeEventListener(listenerId: number): Promise<void> {
+    return await this.logs.removeEventListener(listenerId);
+  }
+
+  /**
+   * Deploy a new contract in a loaded Solang program
+   *
+   * @param program
+   * @param options
+   * @returns
+   */
+  public async deployContract(
+    options: ContractDeployOptions
+  ): Promise<ContractDeployResult> {
+    return Contract.deploy(this, options);
+  }
+
+  /**
+   * Load a deployed contract
+   *
+   * @param program
+   * @param abiData
+   * @param contractStorageAccount
+   * @returns                      The contract instance
+   */
+  public async getContract(
+    abiData: string,
+    contractStorageAccount: Keypair
+  ): Promise<Contract> {
+    return Contract.get(this, abiData, contractStorageAccount);
+  }
```

Comment:
All these methods have one line and just flip some arguments around. It feels like they are adding indirection and API surface area without established cause.